apps/etterlatte-brev-api/src/main/kotlin/no/nav/etterlatte/brev/dokarkiv/DokarkivService.kt

```diff
@@ -25,52 +29,51 @@ class DokarkivServiceImpl(
-        private const val BEHANDLINGSTEMA_BP = "ab0255"
-
-        fun mapTilJournalpostRequest(
-            melding: DistribusjonMelding,
-            dokumentInnhold: ByteArray
-        ): JournalpostRequest = JournalpostRequest(
-            tittel = melding.tittel,
+    private fun mapTilJournalpostRequest(
+        vedtaksbrev: Brev,
+        vedtak: Vedtak,
+        dokumentInnhold: ByteArray
+    ): JournalpostRequest {
+        return JournalpostRequest(
+            tittel = vedtaksbrev.tittel,
             journalpostType = JournalPostType.UTGAAENDE,
             behandlingstema = BEHANDLINGSTEMA_BP,
-            avsenderMottaker = melding.mottaker.tilAvsenderMottaker(),
-            bruker = melding.bruker,
-            eksternReferanseId = "${melding.behandlingId}.${melding.brevId}",
-            sak = Sak(Sakstype.FAGSAK, melding.behandlingId),
-            dokumenter = listOf(dokumentInnhold.tilJournalpostDokument(melding)),
+            avsenderMottaker = vedtaksbrev.mottaker.tilAvsenderMottaker(),
+            bruker = Bruker(vedtak.sak.ident),
+            eksternReferanseId = "${vedtaksbrev.behandlingId}.${vedtaksbrev.id}",
+            sak = Sak(Sakstype.FAGSAK, vedtaksbrev.behandlingId),
+            dokumenter = listOf(dokumentInnhold.tilJournalpostDokument(vedtaksbrev.tittel)),
             tema = "EYB", // https://confluence.adeo.no/display/BOA/Tema
             kanal = "S", // https://confluence.adeo.no/display/BOA/Utsendingskanal
-            journalfoerendeEnhet = melding.journalfoerendeEnhet
+            journalfoerendeEnhet = vedtak.vedtakFattet!!.ansvarligEnhet
         )
     }
-}
 
-private fun Mottaker.tilAvsenderMottaker() = when {
-    foedselsnummer != null -> AvsenderMottaker(id = foedselsnummer!!.value)
-    orgnummer != null -> AvsenderMottaker(id = orgnummer, idType = "ORGNR")
-    adresse != null -> {
-        val mottakerNavn = "${adresse!!.fornavn} ${adresse!!.etternavn}"
-        AvsenderMottaker(id = null, navn = mottakerNavn, idType = null, land = adresse!!.land)
-    }
-    else -> throw Exception("Ingen brevmottaker spesifisert")
-}
+    private fun ByteArray.tilJournalpostDokument(tittel: String) = JournalpostDokument(
+        tittel,
+        brevkode = BREV_KODE,
+        dokumentvarianter = listOf(DokumentVariant.ArkivPDF(Base64.getEncoder().encodeToString(this)))
+    )
 
-private fun ByteArray.tilJournalpostDokument(melding: DistribusjonMelding) = JournalpostDokument(
-    tittel = melding.tittel,
-    brevkode = melding.brevKode,
-    dokumentvarianter = listOf(DokumentVariant.ArkivPDF(Base64.getEncoder().encodeToString(this)))
-)
\ No newline at end of file
+    private fun Mottaker.tilAvsenderMottaker() = when {
+        foedselsnummer != null -> AvsenderMottaker(id = foedselsnummer!!.value)
+        orgnummer != null -> AvsenderMottaker(id = orgnummer, idType = "ORGNR")
+        adresse != null -> {
+            val mottakerNavn = "${adresse!!.fornavn} ${adresse!!.etternavn}"
```

Comment:
Trenger man `!!` når man sjekker for ikke null rett før?

Reply:
Ja, dessverre. Har å gjøre med at `Mottaker` ligger i en annen modul. Kotlin tillater ikke smart casting siden det potensielt kan brekke om modulen/avhengigheten endres og kompileres uten at modulen som bruker den oppdateres. 

Har planer om å flytte common.brev til brev-api siden det nå ikke lenger trenger ligge i common, men vil ta det i egen PR.